src/compression/mod.rs

```diff
@@ -260,10 +261,32 @@ impl Compression {
     pub fn supports_deep_data(self) -> bool {
         use self::Compression::*;
         match self {
-            Uncompressed | RLE | ZIP1 | ZIP16 => true,
+            Uncompressed | RLE | ZIP1 /* | ZIP16*/ => true,
             _ => false,
         }
     }
+
+    /// Most compression methods will reconstruct the exact pixel bytes,
+    /// but some might throw away unimportant data for specific types of samples.
+    pub fn is_lossless_for(self, sample_type: SampleType) -> bool {
+        use self::Compression::*;
+        match self {
+            PXR24 => sample_type != SampleType::F32, // pxr reduces f32 to f24
+            Uncompressed | RLE | ZIP1 | ZIP16 | PIZ => true,
+            B44 | B44A | DWAB | DWAA(_) => false,
```

Comment:
From my understanding of `B44(A)` compression, only `f16` is compressed. The rest is raw copied.

Reply:
Did you have to read the B44 code to find this out, or can I find it somewhere in the C++ comments? :)

Reply:
I find it while translating code. Here is the real code:

compress: https://github.com/AcademySoftwareFoundation/openexr/blob/master/src/lib/OpenEXR/ImfB44Compressor.cpp#L746
decompress: https://github.com/AcademySoftwareFoundation/openexr/blob/master/src/lib/OpenEXR/ImfB44Compressor.cpp#L888

It make sense as B44 is a hand crafted compression of f16 and seems to have no effect on u32/f32.

That's my understanding of the thing.

Reply:
I think you're right, nice. Does this mean that f32 data will be completely uncompressed in the file?!

Also, why is it necessary to copy these channels twice (in to tmp, tmp to out)? We could save one copy and some memory there (just an idea for later, not now).

Reply:
> Edit: Yes it does. Oh no.

More than this: I suspect it's a specific feature to read some channels in real time and having other in full precision. Just a guess but from the original implementation, that's the behavior you could have.

> Also, why is it necessary to copy these channels twice (in to tmp, tmp to out)? We could save one copy and some memory there (just an idea for later, not now).

Yes ! That's the part of the code I need to review. That was my original goal and have difficulties with u8 and u16 array copy so I decided to be full u16 and fall to u8 in a big copy to the end, just to get the "B44 compression working", but its defenetly something to do. Once again, I need more time to mess with slices/types.

Reply:
Ok